### PR TITLE
Initial support for string literals

### PR DESCRIPTION
**Pull request by xbauch**
Chris' PR required only a small push after all.

### CODE REVIEW

File: gnat2goto/driver/tree_walk.adb [hunk 1]
```diff
@@ -1194,6 +1202,43 @@ package body Tree_Walk is
...
+        Make_Signedbv_Type (Ireps.Empty, 64));
+      --  FIXME: Does this need to be a proper signedbv representation, in hex?
+      Set_Value (String_Length_Expr,
+                 Convert_Uint_To_Hex (UI_From_Int (Int (String_Length
+                   (Strval (N)))), 64));
+
+      Set_Type (Ret,
+                Make_Array_Type (
+                  I_Subtype => Element_Type,
+                  Size => String_Length_Expr));
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/driver/tree_walk.adb]: This goes in the right direction I think. Right now the main issue is that this won't match the actual type we assign to string literals...

I'm afraid we'll have to do some fairly annoying trickery with the assignment operator to make that work :(

File: testsuite/gnat2goto/tests/string_literal/test.opt [hunk 2]
```diff
@@ -0,0 +1 @@
+VERIFICATION SUCCESSFUL
```
Comment (hannes-steffenhagen-diffblue) [inline testsuite/gnat2goto/tests/string_literal/test.opt]: This should be `test.out` rather than `test.opt`

**Comment by xbauch** [comment]
Rebased, squashed, removed obsolete fixme comment (one), renamed test output.

File: gnat2goto/driver/tree_walk.adb [hunk 3]
```diff
@@ -1194,6 +1202,42 @@ package body Tree_Walk is
...
+      --  Set_Type (Ret, Bare_Array_Type);
+      --  for J in 1 .. String_Length (Strval (N)) loop
+      --        CC := Get_String_Char (Strval (N), J);
+      --        Litteral_Value (J) := CC;
```
Comment (chrisr-diffblue) [inline gnat2goto/driver/tree_walk.adb]: This comment here can be deleted

Comment (xbauch) [reply gnat2goto/driver/tree_walk.adb]: removed.

**Comment by chrisr-diffblue** [comment]
@xbauch Looks good, thanks - one final request... Could you also add an additional test, xfailed for now, that will actually *use* string literals in some way? The idea being that an xfailed test will remind us to go back and do the re-work/additional work once other parts of gnat2goto are improved enough.

**Comment by xbauch** [comment]
@chrisr-diffblue There's a new (failing) test now (`string_assert`) that checks the length of a string.

File: testsuite/gnat2goto/tests/string_assert/test.out [hunk 4]
```diff
@@ -0,0 +1 @@
+ALL XFAIL CBMC fails types of expressions on each side of equality should match
```
Comment (chrisr-diffblue) [inline testsuite/gnat2goto/tests/string_assert/test.out]: I think this should be in test.opt, not test.out ?

Comment (xbauch) [reply testsuite/gnat2goto/tests/string_assert/test.out]: https://github.com/diffblue/gnat2goto/pull/93#discussion_r242583565 .. now I'm confused.

Comment (chrisr-diffblue) [reply testsuite/gnat2goto/tests/string_assert/test.out]: I just edited my comment about to hopefully clarify - expected passing output goes in 'test.out' and any 'options' about the testcase, such as XFAIL, goes in 'test.opt' See https://github.com/diffblue/gnat2goto/tree/master/testsuite/gnat2goto/tests/recursion for an example.

Comment (xbauch) [reply testsuite/gnat2goto/tests/string_assert/test.out]: @chrisr-diffblue Ok, I think I understand it now.

**Pull request merged**